prac_05/emails.py

```diff
@@ -0,0 +1,31 @@
+"""
+Emails
+Estimate: 15 minutes
+Actual: 20 minutes
+"""
+
+
+def main():
+    """Build dictionary of emails and names"""
+    email_to_name = {}
+    email = input("Email: ")
+    while email != '':
+        name = extract_name_from_email(email)
+        name_check = input(f"Is your name {name}? (Y/n) ").upper()
+        if name_check != 'Y' and name_check != '':
+            name = input("Name: ").title()
+        email_to_name[email] = name
+        email = input("Email: ")
+    for email, name in email_to_name.items():
```

Comment:
Perfect! You can use something like this to do task 4 of the state_names program.